src/Poll.tsx

```diff
@@ -343,13 +343,13 @@ function Poll<TData = any, TError = any, TQueryParams = { [key: string]: any }>(
   // Compose Contexts to allow for URL nesting
   return (
     <RestfulReactConsumer>
-      {contextProps => {
+      {async contextProps => {
         const contextRequestOptions =
           typeof contextProps.requestOptions === "function"
-            ? contextProps.requestOptions()
+            ? await contextProps.requestOptions()
```

Comment:
I saw you didn't add an await to this one in your commits @fabien0102 - was that accidental or did you intend to not await this one?